packages/edit-site/src/components/save-button/index.js

```diff
@@ -78,7 +81,27 @@ export default function SaveButton( {
 		}
 		return __( 'Save' );
 	};
-	const label = getLabel();
+
+	/**
+	 * We focus on adding the customization to the SaveButton's `onClick` and `label` for now.
+	 * We will provide the customization to the other entry points (e.g., SavePanel, SaveHub) in the future if needed.
+	 * @see https://github.com/WordPress/gutenberg/pull/56807
+	 */
+	const onClick = useCallback( () => {
+		const callback = applyFilters( 'edit-site.SaveButton.onClick', () =>
+			setIsSaveViewOpened( true )
+		);
+		callback();
+	}, [ setIsSaveViewOpened ] );
```

Comment:
Should we also add the dependencies here. Did you get a warning? See https://github.com/WordPress/gutenberg/pull/24914